### ISSUE TITLE
RN 0.75.4 new arch

### ISSUE THREAD

**Issue by Mohamed-Ziyad** [react-native-async-storage/async-storage#1156]
### What happened?

Node.js v22.9.0

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-async-storage_async-storage:generateCodegenArtifactsFromSchema'.
> Process 'command 'node'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

### Version

2.0.0

### What platforms are you seeing this issue on?

- [X] Android
- [ ] iOS
- [ ] macOS
- [ ] Windows
- [ ] web

### System Information

```true
My React native info. 

System:
  OS: macOS 14.5
  CPU: (10) arm64 Apple M1 Pro
  Memory: 132.39 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 22.9.0
    path: ~/.nvm/versions/node/v22.9.0/bin/node
  Yarn:
    version: 3.3.1
    path: /opt/homebrew/bin/yarn
  npm:
    version: 10.8.3
    path: ~/.nvm/versions/node/v22.9.0/bin/npm
  Watchman:
    version: 2024.03.25.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.5
      - iOS 17.5
      - macOS 14.5
      - tvOS 17.5
      - visionOS 1.2
      - watchOS 10.5
  Android SDK:
    API Levels:
      - "33"
      - "33"
      - "34"
      - "35"
    Build Tools:
      - 34.0.0
      - 35.0.0
    System Images:
      - android-31 | Intel x86 Atom_64
      - android-33 | Google APIs ARM 64 v8a
      - android-34 | Google APIs ARM 64 v8a
    Android NDK: Not Found
IDEs:
  Android Studio: 2024.1 AI-241.19072.14.2412.12360217
  Xcode:
    version: 15.4/15F31d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.10
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli":
    installed: 15.0.1
    wanted: ^15.0.1
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.75.4
    wanted: 0.75.4
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: true
```

### Steps to Reproduce

- I enabled new arch and try to build I have this issue when I disable it no issue in android 
- I synced with android studio it shows cmake issue
- I update gradle same issue

**Comment by redbluenat** [comment]
+1

**Comment by Makhdoom-Sharif** [comment]
Same issue. @Mohamed-Ziyad  Did you find any workaround for it?

**Comment by Mohamed-Ziyad** [comment]
removed new arch.

**Comment by mensonones** [comment]
Hi! I tried to reproduce the problem, but without success. If you or anyone else is still having this problem and can share a minimal reproduction of it, I would be happy to try to help.
Here is a link to the repository of the project where I tried to reproduce the problem according to the information provided, but I was unable to reproduce it.

Link project: https://github.com/mensonones/ReproduceBugAsyncStorage2

**npx react-native info**

info Fetching system and libraries information...
(node:382558) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
System:
  OS: Linux 6.8 Ubuntu 22.04.5 LTS 22.04.5 LTS (Jammy Jellyfish)
  CPU: (16) x64 11th Gen Intel(R) Core(TM) i7-11800H @ 2.30GHz
  Memory: 10.37 GB / 31.13 GB
  Shell:
    version: 5.1.16
    path: /bin/bash
Binaries:
  Node:
    version: 22.0.0
    path: /usr/local/bin/node
  Yarn:
    version: 1.22.22
    path: /usr/local/bin/yarn
  npm:
    version: 10.5.1
    path: /usr/local/bin/npm
  Watchman:
    version: 4.9.0
    path: /usr/bin/watchman
SDKs:
  Android SDK: Not Found
IDEs:
  Android Studio: Not Found
Languages:
  Java:
    version: 17.0.12
    path: /usr/bin/javac
  Ruby: Not Found
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.75.4
    wanted: 0.75.4
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: Not found
  newArchEnabled: false

info React Native v0.76.1 is now available (your project is running on v0.75.4).
info Changelog: https://github.com/facebook/react-native/releases/tag/v0.76.1
info Diff: https://react-native-community.github.io/upgrade-helper/?from=0.75.4&to=0.76.1
info For more info, check out "https://reactnative.dev/docs/upgrading?os=linux".

**Comment by gelodgreat** [comment]
replicated this issue using RN 0.76 when newArch is enabled by default, is there a way to fix this?

**Comment by Makhdoom-Sharif** [comment]
I have updated the app to 0.75.1 and got the same issue 2 days back but now when I try to reproduce the issue and run the app it works fine with the new arch. I know it's wired 😂, but it is.

**Comment by mensonones** [comment]
hi! @gelodgreat , i have tested on 0.76 with new project and not replicate this this issue.

**Comment by RohanRajGautam** [comment]
Had similar issue when upgrading from v0.74.1 to v0.76.1, after a day of debugging found out that I missed this import line in MainApplication.kt

`import com.facebook.soloader.SoLoader`

**Comment by Mohamed-Ziyad** [comment]
It's fixed, upgrade to version 0.76.1
